Client/Frontend/AuthenticationManager/PasscodeViews.swift

```diff
@@ -179,4 +183,19 @@ class PasscodePane: UIView {
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    
+    func keyboardWillShow(sender: NSNotification) {
+        let keyboardFrame: CGRect = sender.userInfo![UIKeyboardFrameEndUserInfoKey]!.CGRectValue
```

Comment:
Nit: As a general tip, try to avoid using force unwraps if you can since they can often lead to runtime exceptions. I would try writing this using guard and using optionals instead so it's safer.

```swift
guard let keyboardFrame = sender.userInfo?[UIKeyboardFrameEndUserInfoKey]?.CGRectValue else {
    return
}
```